src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -1691,6 +1727,12 @@ public void clearPushRegistrationId() {
         }
 
         @Override
+        public void clearSinglePushRegistrationId(String registrationId) {
+            remove("$android_devices", registrationId);
+            mPersistentIdentity.clearPushId();
```

Comment:
Totally correct, would only want to wipe if the token passed is the same one stored on disk. Changing now.